(FC) Change customer code from email to order number

hugobast opened the pull request:
Here's the [clubhouse card](https://app.clubhouse.io/glossier-dot-com/story/13834/as-a-guest-shopper-i-want-to-be-able-to-use-fc-so-i-can-check-out-as-quickly-as-possible) that has more information about this

Reiterating what I said in a specific commit for context:

```
This change is motivated by a hard to work around flow during ApplePay
checkout flow where we only get the customer's information like their
email at the very end step.

Switching from email to order number still affords us to derive the
customer's information by going from the order to the order's email or
the order's customer and then the email or even addresses.
```

DanielWright commented:
What would the extra lift be to make this configurable? It'd be nice to (eventually) migrate back to the canonical fork, and contribute this back to the community without baking our business rules into the code.

hugobast commented:
Great idea, I'll look into it

hugobast committed:
Make customer code configurable

This change is motivated by a hard to work around flow during ApplePay
checkout flow where we only get the customer's information like their
email at the very end step.

* Fix a failing spec

Validation failed: Payment method can't be blank was getting thrown for
these 2 tests, and yes checking the payment class source file correctly
points out that this is required. Invoking the payment factory is a nice
solution to this.

DanielWright commented on app/models/spree_avatax/config.rb:
```diff
@@ -25,11 +27,25 @@ def enabled
+      end
+
+      def customer_code=(evaluator)
+        raise(Exception) unless evaluator.respond_to?(:call)
```
I might use a more descriptive name for this exception, for readability's sake.

DanielWright replied:
Indeed, what do you think of avoiding an exception altogether by wrapping the argument _in_ a proc if it's not callable?

DanielWright reviewed (approved):
Muy bueno. Just a small suggestion/inquiry.

hugobast committed:
Remove need for an error

We wrap the value into a proc as to make it conform to what the gem
expects internally.
